From 71851c72207f8db15c690133d7f7d0ae8bedec6c Mon Sep 17 00:00:00 2001
From: VinceChang6637 <vince_chang@aspeedtech.com>
Date: Fri, 21 Jun 2024 16:56:52 +0800
Subject: [PATCH] revert limit to on chunk

Revert limit to on chunk commit.
https://github.com/openbmc/webui-vue/commit/740cbd525fa619e6a4190ec08c80f45ae72435eb

We need the "0001-Update-to-vue-5.0.8.patch" to upgrade vue/cli to version 5.0.8.
However, using "LimitChunkCountPlugin " causes the following build error:
```
Cannot read properties of undefined (reading 'tap')
```

This issue is due to differences between webpack 4 and 5.
```
$ npm list webpack
├─┬ @vue/cli-plugin-babel@5.0.8
│ └── webpack@5.92.1
├─┬ @vue/cli-plugin-eslint@5.0.8
│ └── webpack@5.92.0
├─┬ @vue/cli-service@5.0.8
│ └── webpack@5.92.1
├─┬ react-scripts@5.0.1
│ └── webpack@5.92.1
└─┬ vuepress@1.8.2
  └─┬ @vuepress/core@1.8.2
    └── webpack@4.47.0
```

This patch is a workaround to avoid build failures.

Upstream-Status: Inappropriate [ASPEED specific]
Signed-off-by: VinceChang6637 <vince_chang@aspeedtech.com>
---
 vue.config.js | 20 --------------------
 1 file changed, 20 deletions(-)

diff --git a/vue.config.js b/vue.config.js
index d073335..598ca43 100644
--- a/vue.config.js
+++ b/vue.config.js
@@ -1,6 +1,4 @@
 const CompressionPlugin = require('compression-webpack-plugin');
-const webpack = require('webpack');
-const LimitChunkCountPlugin = webpack.optimize.LimitChunkCountPlugin;
 
 module.exports = {
   css: {
@@ -75,16 +73,6 @@ module.exports = {
     }
   },
   configureWebpack: (config) => {
-    config.plugins.push(
-      new LimitChunkCountPlugin({
-        maxChunks: 1,
-      }),
-    );
-    config.optimization.splitChunks = {
-      cacheGroups: {
-        default: false,
-      },
-    };
     const crypto = require('crypto');
     const crypto_orig_createHash = crypto.createHash;
     crypto.createHash = (algorithm) =>
@@ -129,14 +117,6 @@ module.exports = {
     config.experiments = {
       asyncWebAssembly: true,
     };
-
-    config.performance = {
-      hints: 'warning',
-      maxEntrypointSize: 512000,
-      maxAssetSize: 512000,
-    };
-
-    config.optimization.runtimeChunk = false;
   },
   pluginOptions: {
     i18n: {
